ansible_base/lib/utils/encryption.py

```diff
@@ -37,34 +38,62 @@ def __init__(self):
         self._encryption_key = self.key[32:]
         self._backend = default_backend()
 
+    def extract_data(self, raw_data: str) -> str:
+        # Sequential data extract for code reuse in is_encrypted_string and decrypt_string
+        raw_data = raw_data[len(ENCRYPTED_STRING) :]
```

Comment:
I noticed that we are not doing the `if raw_data.startswith(ENCRYPTED_STRING)` here. Which is ok for how you have the function setup but would not be ok if we called this directly and the string was missing it. I think we should add that here to protect someone from calling this incorrectly. What are your thoughts?